api/controllers/upgrade.js

```diff
@@ -1,68 +1,23 @@
-const PouchDB = require('pouchdb-core');
-PouchDB.plugin(require('pouchdb-adapter-http'));
-
-const buildDbUrl = 'https://staging.dev.medicmobile.org/_couch/builds';
-
-const buildDb = new PouchDB(buildDbUrl);
-const targetDbUrl = process.env.COUCH_URL;
-const targetDb = new PouchDB(targetDbUrl);
-
-const stagedDdocName = name => name + ':staged';
-
-const ddocName = buildInfo =>
- `${buildInfo.namespace}:${buildInfo.application}:${buildInfo.version}`;
-
-module.exports = (buildInfo, username) => {
-  if (!buildInfo) {
-    throw new Error('Bad request');
-  }
-
-  // While we've set the api to in theory be more generic, this is for if we
-  // split this code out (and say, put it with horticulturalist). For now we
-  // just support medic deploying medic :-)
-  if (buildInfo.namespace !== 'medic' ||
-      buildInfo.application !== 'medic') {
-    return Promise.reject({
-      status: 400,
-      message: 'We only support medic as the application and namespace'
-    });
-  }
-
-  console.log('upgrade()', `Upgrading to ${JSON.stringify(buildInfo)}…`);
-
-  console.log('upgrade()', 'Fetching newDdoc…');
-  return buildDb
-    .get(ddocName(buildInfo), { attachments:true })
-    .catch(err => {
-      if (err.status === 404) {
-        err = new Error(`Version not found: ${buildInfo.version}`);
-      }
-
-      throw err;
-    })
-    .then(newDdoc => {
-      console.log('upgrade()', 'Fetched newDdoc.');
-
-      console.log('upgrade()', 'Fetching oldDdoc…');
-      return targetDb.get('_design/medic')
-        .then(oldDdoc => {
-          console.log('upgrade()', 'Fetched oldDdoc.');
-
-          newDdoc._id = stagedDdocName(oldDdoc._id);
-          delete newDdoc._rev;
-
-          newDdoc.deploy_info = {
-            timestamp: new Date().toString(),
-            user: username,
-            version: buildInfo.version,
+const auth = require('../auth'),
+      serverUtils = require('../server-utils');
+
+const service = require('../services/upgrade');
+
+module.exports = {
+  upgrade: (req, res) => {
+    return auth.check(req, '_admin')
```

Comment:
I think this is inconsistent with the UI - non-db-admins with the "can_configure" permission will be able to see the upgrade button but get an error when they click it. Either add a new permission for "can_update_app" (or a decent name) and use that in the front end too, or change this to use "can_configure".

Reply:
Yeah good point. I think I'll change it to `can_configure` for now… I think that makes sense, in the future we can always restrict it further.